packages/java/src/node/java-service.ts

```diff
@@ -0,0 +1,69 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { NodePackage, ExtensionPackage } from '@theia/application-package';
+import { JavaService } from '@theia/java/src/common/java-common';
+import { ApplicationPackageManager } from '@theia/application-manager';
+import { injectable } from 'inversify';
+import * as paths from 'path';
+
+interface JavaContributes extends NodePackage {
+    javaExtensions?: string[];
+}
+
+/**
+ * JavaService implementation.
+ */
+@injectable()
+export class JavaServiceImpl implements JavaService {
+    private javaExtensions: string[];
+
+    async collectJavaExtensions(): Promise<string[]> {
+        if (this.javaExtensions) {
+            return Promise.resolve(this.javaExtensions);
+        }
+
+        this.javaExtensions = [];
+        const projectPath = process.cwd();
+        const manager = new ApplicationPackageManager({ projectPath });
+        const extensions = manager.pck.extensionPackages;
+        extensions.forEach(extensionInfo => {
+            this.findBundles(extensionInfo);
+        });
+        return this.javaExtensions;
+    }
+
+    private findBundles(extPck: ExtensionPackage) {
+        const contributes: JavaContributes = extPck.raw.contributes;
+        if (!contributes) {
+            return;
+        }
+        const javaExtensions = contributes.javaExtensions;
+        if (Array.isArray(javaExtensions) && javaExtensions.length) {
+            if (!extPck.raw.installed) {
+                return;
+            }
+            const packagePath = extPck.raw.installed.packagePath;
+            for (const javaExtensionPath of javaExtensions) {
+                this.javaExtensions.push(paths.resolve(packagePath, '..', javaExtensionPath));
+            }
+        }
+    }
+
+    dispose(): void {
```

Comment:
`JavaService` don't need to be `Disposable` or a method should not do anything, otherwise, it can throw if someone call `dispose`

Reply:
fixed, thank you!